src/features/README.md

```diff
@@ -82,6 +86,75 @@ earthengine:
+
+Each of the 37 time windows corresponds to a single satellite image, which is a mosaic over that particular 10 day interval.
+
+### Sentinel-2 Mosaic Retrieval
+
+The satellite imagery we use for the time series is Sentinel-2 L1C data (available starting June 2015), retrieved through [Google Earth Engine](https://developers.google.com/earth-engine/datasets/catalog/COPERNICUS_S2_HARMONIZED)
+
+For each of the 37 ten-day windows in the time series, we generate a mosaic image spanning the interval, and save it to our Google Cloud Bucket. Each of these resultant images contains 13 bands: all 10 original Sentinel-2 bands and 3 derived bands to measure vegetation.
+
+Preprocessing for Each Mosaic:
+
+- Pseudo-Atmospheric Correction (DOS):
+    - Each Sentinel-2 L1C mosaic is first corrected using a simple Dark Object Subtraction (DOS)algorithm. This reduces atmospheric haze and brings the reflectance values closer to L2A surface reflectance.
+
+- Custom Scene Classification Layer (SCL):
+    - After atmospheric correction, a custom SCL (Scene Classification Layer) band is generated based on NDVI, NDWI, NDSI, and brightness thresholds. This SCL simulates the L2A official product, enabling masking of clouds, shadows, water, snow/ice, vegetation, and more, and allows downstream analysis to use L2A-like quality masks for each time step.
+
+Each of these resultant images contains the following bands:
+
+- **10 Original Sentinel-2 Bands**: B2, B3, B4, B5, B6, B7, B8, B8A, B11, B12
+- **NDVI: Normalized Difference Vegetation Index**: Measures green vegetation density, computed from NIR and Red bands (B8, B4).
+
+$$
+\text{NDVI} = \frac{\text{NIR} - \text{Red}}{\text{NIR} + \text{Red}}
+$$
+
+- **EVI: Enhanced Vegetation Index**: Similar to NDVI, but slightly better in areas with dense canopy or haze. Computed from NIR, Red, and Blue bands (B8, B4, B2)
+
+$$
+\text{EVI} = 2.5 \times \frac{\text{NIR} - \text{Red}}{\text{NIR} + 6 \times \text{Red} - 7.5 \times \text{Blue} + 1}
+$$
+
+- **NDWI: Normalized Difference Water Index**: Detects moisture changes in vegetation and soil. Computed from NIR and SWIR1 bands (B8, B11).
+
+$$
+\text{NDWI} = \frac{\text{NIR} - \text{SWIR}}{\text{NIR} + \text{SWIR}}
+$$
+
+- **Custom SCL Band**: Scene classification for cloud, shadow, water, vegetation, etc.
+
+#### Handling Missing and Invalid Data
+
+For a particular window, data may be missing (if there is no satellite imagery within that timeframe) or invalid (if there are clouds covering the image)
+
+- **Cloud Detection** After retrieving all bands, we use module `s2cloudless` to detect pixels affected by clouds or cloud shadows. These pixels are set to -9999 across all bands to indicate missing/invalid data.
+
+- **Missing Images** In rare cases, a time window may have no available satellite imagery. When this occurs, all pixels are assigned a value of -9999 to indicate missing data. Additionally, we set `cloud_fraction = 1.0` in the corresponding metadata file.
```

Comment:
Do we have any idea how frequent this is? It would be nice, once we have gone through and downloaded the images, to get some stats on how frequently this happens as a sanity check